api/api_signer.go

```diff
@@ -73,6 +73,33 @@ func NewSignHandlerFromSigner(signer signer.Signer) (h HTTPHandler, err error) {
 	}, nil
 }
 
+// a type meant to be unmarshalled from JSON so that there can be a
```

Comment:
A nit, but can you make this a complete sentence? It'll be easier to read and understand later on.